controllers/operatorscaler_controller.go

```diff
@@ -177,6 +186,48 @@ func (r *OperatorScalerReconciler) isOdfDependenciesCsvReady() error {
 	return nil
 }
 
+func (r *OperatorScalerReconciler) createMetrics() error {
+	r.log.Info("entering createMetrics")
+
+	var returnErr error
+
+	for i := range ResourceMappingList {
+		resMap := &ResourceMappingList[i]
+
+		if resMap.ReportMetrics {
+
+			crList := &metav1.PartialObjectMetadataList{}
+			crList.TypeMeta.APIVersion = resMap.ApiVersion
+			crList.TypeMeta.Kind = resMap.Kind
+
+			if err := r.Client.List(r.ctx, crList); err != nil {
+				if !meta.IsNoMatchError(err) {
+					msg := fmt.Sprintf("failed listing %s", resMap.Kind)
+					r.log.Error(err, msg)
+					multierr.AppendInto(&returnErr, err)
+				}
+			} else if err == nil {
+				for j := range crList.Items {
+					crItem := &crList.Items[j]
+
+					metrics.ReportODFSystemMapMetrics(
```

Comment:
may I know what type of prom metrics is this? are we fine w/ just exporting this during reconcile?

Reply:
This is the same metrics that UI folks were talking about, and even earlier, we were reporting inside the reconcile.